ffn/utils/proofreading.py

```diff
@@ -182,58 +260,293 @@ def __init__(self, objects, bad, num_to_prefetch=10, locations=None):
+      mode: Type of the error ('merger' or 'split').
+      idx: Indicates if it's the first or second coordinate (0 or 1).
+    """
+    location = self.get_cursor_position(action_state)
+    if location is None:
+      return
+
+    if idx == 1 and not self.temp_coord_list:
+      self.update_msg("You have not entered a first coord yet")
+      return
+
+    if idx == 0 and self.temp_coord_list:
+      self.temp_coord_list = []
+
+    self.temp_coord_list.append(location)
+
+    if idx == 1:
+      if self.temp_coord_list[0] == self.temp_coord_list[1]:
+        self.update_msg("You entered the same coordinate twice. Try again!")
+        self.temp_coord_list = []
+        return
+
+      identifier = self.get_id(mode=mode)
+      self.seg_error_coordinates.update({identifier: self.temp_coord_list})
+      self.annotate_error_locations(self.temp_coord_list, identifier)
+      self.temp_coord_list = []
+
+  def annotate_error_locations(
+      self, coordinate_lst: list[list[int]], id_: str
+  ) -> None:
+    """Annotate the error locations in the viewer.
+
+    Args:
+      coordinate_lst: List of coordinates to be annotated.
+      id_: Unique identifier for the error.
+    """
+    for i, coord in enumerate(coordinate_lst):
+      annotation_id = id_ + f"_{i}"
+      self.mk_point_annotation(coord, annotation_id)
+
+  def mk_point_annotation(
+      self, coordinate: list[int], annotation_id: str
+  ) -> None:
+    """Create a point annotation in the viewer.
+
+    Args:
+      coordinate: 3D coordinate of the annotation point.
+      annotation_id: Unique identifier for the annotation.
+    """
+    if annotation_id.startswith("m"):
+      color = "#fae505"
+    else:
+      color = "#05f2fa"
+    annotation = neuroglancer.PointAnnotation(
+        id=annotation_id, point=coordinate, props=[color]
+    )
+    with self.viewer.txn() as s:
+      annotations = s.layers["annotation"].annotations
+      annotations.append(annotation)
+
+  def get_annotation_id(
+      self, action_state: neuroglancer.viewer_config_state.ActionState
+  ) -> Optional[str]:
+    """Retrieve the ID of a selected annotation.
+
+    Args:
+      action_state: neuroglancer.viewer_config_state.ActionState.
+
+    Returns:
+      The selected object's ID or None if retrieval fails.
+    """
+    try:
+      selection_state = action_state.selected_values["annotation"].to_json()
+      selected_object = selection_state["annotationId"]
+    except Exception:
+      self.update_msg("Could not retrieve annotation id")
+      return
+
+    return selected_object
+
+  def delete_location_from_annotation(
+      self, action_state: neuroglancer.viewer_config_state.ActionState
+  ) -> None:
+    """Delete the error location pair associated with the annotation at the cursor position
+
+    Args:
+      action_state: State of the viewer during the action.
+    """
+    id_ = self.get_annotation_id(action_state)
+    if id_ is None:
+      return
+
+    target_key = id_[:2]
```

Comment:
IIUC, the IDs here will be something m0_0, s1_0, etc. What happens if there are more than 10 splits/mergers?

Reply:
it will break/delete the wrong annotation. Thanks! I'll fix that!